extension-common/pxfconfigparser.c

```diff
@@ -0,0 +1,538 @@
+ * str will be modified
+ */
+static void
+trim_space(char *str)
+{
+    // Find the index of the first non-space char
+    int start = 0;
+    while (isspace(str[start]))
+    {
+        start++;
+    }
+
+    if (str[start] == '\0')
+    {
+        // The string only has spaces
+        str[0] = '\0';
+        return;
+    }
+
+    // Find the index of the last non-space char
+    int end = (int)strlen(str) - 1;
+    while (end >= start && isspace(str[end]))
+    {
+        end--;
+    }
+
+    // Shift all chars to the beginning of the string
+    int len = end - start + 1;
+    for (int i = 0; i < len; i++)
+    {
+        str[i] = str[start + i];
+    }
+
+    // Terminate the string
+    str[len] = '\0';
+}
+
+/*
+ * Parse the tokens in the string separated by delim
+ * str will be modified for tokenizing
+ * Return NULL if any token is empty
+ * Return a List of trimmed tokens where token pointers will point to the indices in the original str
+ * If str is a local string stored on stack, DO NOT re-use the tokens after the caller function returns
+ * because they will be wild pointers once the original string ends its life
+ * If re-use is necessary, make a copy using strdup and free it later
+ */
+static List *
+get_tokens(char *str, char *delim)
+{
+    List *tokens = NULL;
+    char *save_ptr = NULL; // Save the progress of the tokenizer
+
+    for (char *token = strtok_r(str, delim, &save_ptr);
+         token != NULL;
+         token = strtok_r(NULL, delim, &save_ptr))
+    {
+        trim_space(token);
+        if (strlen(token) == 0)
+        {
+            // Empty token is not allowed
+            return NULL;
+        }
+        tokens = lappend(tokens, token);
+    }
+
+    return tokens;
+}
+
+/*
+ * Parse a string into an int
+ * key is used for error message only
+ */
+static int parse_int(const char *str, const char *key)
+{
+    char *end_ptr;
+    long num = strtol(str, &end_ptr, 10);
+    if (errno == ERANGE || num != (long)(int)num)
+    {
+        // Out of long range during strtol function call
+        // Or will out of range if converting long to int
+        parser_terminate("configuration property '%s' is out of integer range", key);
+    }
+    if ((errno != 0 && num == 0) || *end_ptr != '\0' || end_ptr == str)
+    {
+        // There are another three types of failures
+        // 1. Unspecified failures
+        // 2. Conversion is not finished
+        // 3. Conversion is not started
+        // The unfinished conversion can be because of trailing characters such as "123abc"
+        // The not started conversion can be because of empty string
+        parser_terminate("configuration property '%s' is set to an invalid integer value", key);
+    }
+    return (int)num;
+}
+
+/*
+ * Parse a string into a bool
+ * key is used for error message only
+ */
+static bool parse_bool(const char *str, const char *key)
+{
+    if (strcmp(str, "true") == 0 || strcmp(str, "1") == 0)
```

Comment:
I don't think 1 and 0 are valid representation of boolean values for our properties
```suggestion
    if (strcmp(str, "true") == 0)
```